src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -237,13 +256,116 @@ public static class AkkaClusterHostingExtensions
+
+                var singletonProxySettings =
+                    ClusterSingletonProxySettings.Create(system).WithSingletonName(singletonName);
+
+                if (!string.IsNullOrEmpty(options.Role))
+                {
+                    clusterSingletonManagerSettings = clusterSingletonManagerSettings.WithRole(options.Role);
+                    singletonProxySettings = singletonProxySettings.WithRole(options.Role);
+                }
+
+                var singletonProps = options.TerminationMessage == null
+                    ? ClusterSingletonManager.Props(actorProps, clusterSingletonManagerSettings)
+                    : ClusterSingletonManager.Props(actorProps, options.TerminationMessage,
+                        clusterSingletonManagerSettings);
+
+                var singletonManagerRef = system.ActorOf(singletonProps, singletonName);
+
+                // create a proxy that can talk to the singleton we just created
+                // and add it to the ActorRegistry
+                if (createProxyToo)
+                {
+                    if (options.BufferSize != null)
+                    {
+                        singletonProxySettings = singletonProxySettings.WithBufferSize(options.BufferSize.Value);
+                    }
+
+                    CreateAndRegisterSingletonProxy<TKey>($"/user/{singletonManagerRef.Path.Name}", singletonManagerRef.Path.Name, singletonProxySettings, system, registry);
+                }
+            });
+        }
+
+        private static void CreateAndRegisterSingletonProxy<TKey>(string singletonActorName, string singletonActorPath,
+            ClusterSingletonProxySettings singletonProxySettings, ActorSystem system, IActorRegistry registry)
+        {
+            var singletonProxyProps = ClusterSingletonProxy.Props(singletonActorPath,
+                singletonProxySettings);
+            var singletonProxy = system.ActorOf(singletonProxyProps, $"{singletonActorName}-proxy");
+            
+            registry.Register<TKey>(singletonProxy);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="ClusterSingletonProxy"/> and adds it to the <see cref="ActorRegistry"/> using the given
+        /// <see cref="TKey"/>.
+        /// </summary>
+        /// <param name="builder">The builder instance being configured.</param>
+        /// <param name="singletonName">The name of this singleton instance. Will also be used in the <see cref="ActorPath"/> for the <see cref="ClusterSingletonManager"/> and
+        /// optionally, the <see cref="ClusterSingletonProxy"/> created by this method.</param>
+        /// <param name="options">Optional. The set of options for configuring the <see cref="ClusterSingletonProxy"/>.</param>
+        /// <param name="singletonManagerPath">Optional. By default Akka.Hosting will assume the <see cref="ClusterSingletonManager"/> is hosted at "/user/{singletonName}" - but
```

Comment:
XML-DOC comment explains it all, but this is mostly for backwards compatibility with applications that created singletones without Akka.Hosting and thus the `ClusterSingletonManager` names were less standardized.